clang/test/SemaSYCL/msvc-macro.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %clang -fsycl -c %s
```

Comment:
I think the check should be done differently.
Ideally, we should do cross-compilation for Windows, currently test detects the native OS, so people running this test on Linux won't be able to detect regressions.

Probably it's better to move pre-processor tests. E.g. improve this test: https://github.com/intel/llvm/blob/sycl/clang/test/Preprocessor/sycl-macro.cpp

```
RUN: %clang_cc1 %s -fsycl-is-device -E -dM -aux-triple=...-windows-msvc | FileCheck --check-prefix=CHECK-MSVC %s

CHECK-MSVC-NOT: __GNUC__
CHECK-MSVC-NOT: __STDC__
```